### PR TITLE
machines: Replace tooltips with Patternfly's FieldLevelHelp

### PR DESCRIPTION
**Pull request by skobyda**
And remove InfoRecord component, which was used only in vcpuModal.

With Patternfly's tooltips instead of our custom ones:
Screenshot from 2019-09-17 13-04-17
Screenshot from 2019-09-17 13-04-34

### CODE REVIEW

**Comment by KKoukiou** [comment]
@andreasn or @garrett please approve the move to the PF-4 component. I will review the code but this introduces a difference on how the infotext is presented.

**Comment by marusak** [comment]
Just for the record of what we talked on todays stand-up - This is not PF4 component but PF3 component. And there is no PF4 equivalent to this PF3 component.

It would be great if we could do this move in one step straight to PF4 component. @skobyda what do you think? Will you try to do that or you prefer to land this as well?

**Comment by garrett** [comment]
> It would be great if we could do this move in one step straight to PF4 component.

I agree with this.

I'd only suggest using PF3 components instead of either PF4 ones or our custom ones if:

- PF4 isn't available
- PF3 is accessible and works well (although it _may_ need to be reskinned to match)
- our component is not accessible
- our component is old or buggy (in this case, a fix could work too)

Examples:
- We used the PF3 typeahead find because
   - the PF4 one is buggy, incomplete, and doesn't support features we need
   - the PF3 one was already way better than our custom component (with keyboard navagatability, etc.)
- We're going to keep using our custom switch for the time being
   - It was implemented recently to be much better, more accssible, and match PF4.
   - We onsidered the PF4 widget, but it has an awful focus ring that's even active on mouse clicks. Ours has a much better focus ring and it is intentionally not triggered by mouse clicks, only keyboard nav.

_**Summary**: Unless there's an exceptional reason, place try to use the PF4 components/widgets, not PF3 ones._

**Comment by marusak** [comment]
PF4 components that could work:
[tooltip](https://pf4.patternfly.org/components/Tooltip/examples/) - classical on hover tooltip
[popover](https://pf4.patternfly.org/components/Popover/examples/) - click to show and then click again to close

**Comment by skobyda** [comment]
**Tooltip** windows has some background color problems (there are 2 background colors, black in middle of window and grey on borders).
**Popover** doesn't work in PF3 Modal, so some bigger overhaul of VCPUModal would be needed.

I wanted this PR as a quick replacement for our old custom component, which is not used anywhere outside of vcpuModal. But now I see that best solution would be to replace these custom tooltips with Popover component together with replacing PF3 modal with PF4 modal to make it work. So this change needs to be part of some bigger move from PF3->PF4.

**Comment by garrett** [comment]
@skobyda: What to you mean about tooltip color problems? Is there a bug? (If so, it should be fixed.)

**Comment by skobyda** [comment]
Oh nevermind, it was problem on my side. I had to rebuild from clean repo and color problem is gone. So we can safely use PF4 Tooltip. (sorry for closing and reopening :( ).

**Comment by skobyda** [comment]
Replaced it with PF4 Tooltip. Design is same as with PF3 FieldLevelHelp.
Screenshot from 2019-09-19 14-43-09

File: pkg/machines/components/vcpuModal.jsx [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Button, Modal } from 'patternfly-react';
+import { Modal, Button } from 'patternfly-react';
```
Comment (KKoukiou) [inline pkg/machines/components/vcpuModal.jsx]: Irrelevant change

**Comment by martinpitt** [comment]
Honestly I like our current semi-transparent gray bootstrap tooltips a lot more than the solid black PF4 ones. They look a lot less bulky, and you can still see what they cover. Maybe that was done for a11y reasons? This isn't a veto, just a "meh". But if we do this, we should (medium-term) change all our tooltips, so that pages don't become inconsistent.

**Comment by andreasn** [comment]
> Honestly I like our current semi-transparent gray bootstrap tooltips a lot more than the solid black PF4 ones. They look a lot less bulky, and you can still see what they cover. 

I think they are OK, but if critical enough of an issue, it should be filed against Patternfly and solved on the toolkit level. Just annoying to carry around tons of custom css for little gain.
https://github.com/patternfly/patternfly-next/issues

**Comment by marusak** [comment]
> But if we do this, we should (medium-term) change all our tooltips, so that pages don't become inconsistent.

Good point. @skobyda do you think it is too big task for this PR or is it not that bad and it could land in this PR as well?

**Comment by martinpitt** [comment]
Changing all our tooltips should be its own set of PRs, it's easier to do it page by page.

**Pull request merged**